src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -249,18 +249,25 @@ pub trait TimestampProvider {
 
         // In order to use a timestamp oracle, we must be in the context of some timeline. In that
         // context we would use the timestamp oracle in the following scenarios:
-        // - The isolation level is Strict Serializable and the `when` allows us to use the
-        //   the timestamp oracle (ex: queries with no AS OF).
+        // - The isolation level is Strict Serializable or Strong Session Serializable and the
+        //   `when` allows us to use the timestamp oracle (ex: queries with no AS OF).
+        // - The isolation level is Serializable, the `when` allows us to use the timestamp oracle
+        //   (ex: queries with no AS OF), and the timeline context is timestamp dependent (ex:
+        //   static queries that have a temporal filter).
         // - The `when` requires us to use the timestamp oracle (ex: read-then-write queries).
+        let advance_non_serializable = when.can_advance_to_timeline_ts()
+            && matches!(
+                isolation_level,
+                IsolationLevel::StrictSerializable | IsolationLevel::StrongSessionSerializable
+            );
+        let advance_serializable = when.can_advance_to_timeline_ts()
+            && isolation_level == &IsolationLevel::Serializable
```

Comment:
Same here: shouldn't this happen for any level weaker or equal to `Serializable`?